packages/driver/cypress/integration/e2e/multi-domain/navigation_spec.ts

```diff
@@ -0,0 +1,108 @@
+import { assertLogLength } from '../../../support/utils'
+
+// @ts-ignore / session support is needed for visiting about:blank between tests
+describe('navigation', { experimentalSessionSupport: true, experimentalMultiDomain: true }, () => {
+  let logs: any = []
+
+  describe('navigation events', () => {
+    beforeEach(() => {
+      logs = []
+
+      cy.on('log:added', (attrs, log) => {
+        logs.push(log)
+      })
+
+      cy.visit('/fixtures/multi-domain.html')
+      cy.get('a[data-cy="multi-domain-secondary-link"]').click()
+    })
+
+    it('navigation:changed via hashChange', (done) => {
+      cy.switchToDomain('foobar.com', done, () => {
+        Cypress.once('navigation:changed', () => {
+          expect(location.host).to.equal('foobar.com')
```

Comment:
Suggest verifying the hash.
```suggestion
          expect(location.hash).to.equal('#hashChange')
```

Reply:
updated